src/test/java/org/folio/entlinks/controller/delegate/InstanceAuthorityStatServiceDelegateTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@UnitTest
+@ExtendWith(MockitoExtension.class)
+class InstanceAuthorityStatServiceDelegateTest {
+
+  private @Mock AuthorityDataStatService statService;
+  private @Mock AuthorityDataStatMapper mapper;
+  private @Mock UsersClient usersClient;
+
+  private @InjectMocks InstanceAuthorityStatServiceDelegate delegate;
+
+  @BeforeEach
+  void setUp() {
+    delegate = new InstanceAuthorityStatServiceDelegate(statService, mapper, usersClient);
+  }
+
+  @Test
+  void fetchStats() {
+    var userIds = List.of(UUID.randomUUID(), UUID.randomUUID());
+    var statData = TestUtils.dataStatList(userIds.get(0), userIds.get(1));
+    var users = TestUtils.usersList(userIds);
+
+    var fromDate = OffsetDateTime.of(2022, 10, 10, 15, 30, 30, 0, ZoneOffset.UTC);
+    var toDate = OffsetDateTime.now();
+    var dataStatActionDto = AuthorityDataStatActionDto.UPDATE_HEADING;
+
+    when(statService.fetchDataStats(fromDate, toDate, dataStatActionDto, 3)).thenReturn(statData);
+    when(usersClient.query(anyString())).thenReturn(users);
+    AuthorityDataStat authorityDataStat1 = statData.get(0);
+    AuthorityDataStat authorityDataStat2 = statData.get(1);
+    var userList = users.getResult();
+    when(mapper.convertToDto(authorityDataStat1))
+      .thenReturn(TestUtils.getStatDataDto(authorityDataStat1, userList.get(0)));
+    when(mapper.convertToDto(authorityDataStat2))
+      .thenReturn(TestUtils.getStatDataDto(authorityDataStat2, userList.get(0)));
+
+    var authorityChangeStatDtoCollection = delegate.fetchAuthorityLinksStats(
+      fromDate,
+      toDate,
+      dataStatActionDto,
+      2
+    );
+
+    assertNotNull(authorityChangeStatDtoCollection);
+    assertNotNull(authorityChangeStatDtoCollection.getStats());
+    assertEquals(2, authorityChangeStatDtoCollection.getStats().size());
+    var resultStatDtos = authorityChangeStatDtoCollection.getStats();
+    for (AuthorityDataStatDto statDto : resultStatDtos) {
+      assertNotNull(statDto.getAction());
+      assertNotNull(statDto.getAuthorityId());
+      assertNotNull(statDto.getHeadingNew());
+      assertNotNull(statDto.getHeadingOld());
+      assertNotNull(statDto.getHeadingTypeNew());
+      assertNotNull(statDto.getHeadingTypeOld());
+      assertNotNull(statDto.getLbFailed());
+      assertNotNull(statDto.getLbTotal());
+      assertNotNull(statDto.getLbUpdated());
+      assertNotNull(statDto.getMetadata());
+      assertNotNull(statDto.getMetadata().getStartedByUserId());
+      assertNotNull(statDto.getMetadata().getStartedByUserFirstName());
+      assertNotNull(statDto.getMetadata().getStartedByUserLastName());
+      assertNotNull(statDto.getNaturalIdNew());
+      assertNotNull(statDto.getNaturalIdOld());
+      assertNotNull(statDto.getSourceFileNew());
+      assertNotNull(statDto.getSourceFileOld());
```

Comment:
What about dates in metadata? They should also be populated.

Reply:
done!